sdk/enoki/src/wallet/index.ts

```diff
@@ -0,0 +1,341 @@
+	},
+];
+
+export class EnokiWallet implements Wallet {
+	#events: Emitter<WalletEventsMap>;
+	#accounts: ReadonlyWalletAccount[];
+	#name: string;
+	#icon: Wallet['icon'];
+	#flow: EnokiFlow;
+	#provider: AuthProvider;
+	#clientId: string;
+	#redirectUrl: string | undefined;
+	#network: EnokiNetwork;
+	#sponsor;
+	#client;
+
+	get name() {
+		return this.#name;
+	}
+
+	get icon() {
+		return this.#icon;
+	}
+
+	get version() {
+		return '1.0.0' as const;
+	}
+
+	get chains() {
+		return [`sui:${this.#network}`] as const;
+	}
+
+	get accounts() {
+		return this.#accounts;
+	}
+
+	get features(): StandardConnectFeature &
+		StandardDisconnectFeature &
+		StandardEventsFeature &
+		SuiSignTransactionFeature &
+		SuiSignAndExecuteTransactionFeature &
+		SuiSignPersonalMessageFeature {
+		return {
+			'standard:connect': {
+				version: '1.0.0',
+				connect: this.#connect,
+			},
+			'standard:disconnect': {
+				version: '1.0.0',
+				disconnect: this.#disconnect,
+			},
+			'standard:events': {
+				version: '1.0.0',
+				on: this.#on,
+			},
+			'sui:signTransaction': {
+				version: '2.0.0',
+				signTransaction: this.#signTransaction,
+			},
+			'sui:signAndExecuteTransaction': {
+				version: '2.0.0',
+				signAndExecuteTransaction: this.#signAndExecuteTransaction,
+			},
+			'sui:signPersonalMessage': {
+				version: '1.0.0',
+				signPersonalMessage: this.#signPersonalMessage,
+			},
+		};
+	}
+
+	constructor({
+		name,
+		icon,
+		flow,
+		provider,
+		clientId,
+		redirectUrl,
+		client,
+		network,
+		sponsor,
+	}: {
+		icon: Wallet['icon'];
+		name: string;
+		flow: EnokiFlow;
+		provider: AuthProvider;
+		clientId: string;
```

Comment:
I think you could use this to define the `id` on the wallet so that you technically could register two google wallets

Reply:
done